tests/Checker/ReferenceAssignmentCheckerTest.php

```diff
@@ -106,6 +106,20 @@ public function testSure()
         $this->thenIShouldReceiveNoWarning();
     }
 
+    public function testBlocks()
```

Comment:
I see the visitor as an implementation detail that should be tested in context. We could test it additionally, but I would not move those tests.